src/features/filter.ts

```diff
@@ -1,10 +1,66 @@
-export const actions = { /* put action creators here */};
+type AllAction = { type: 'filter/ALL'; };
+type ActiveAction = { type: 'filter/ACTIVE'; };
+type CompletedAction = { type: 'filter/COMPLETED'; };
+type ByQueryAction = {
+  type: 'filter/QUERY',
+  payload: string
+};
+
+type Action = AllAction
+| ActiveAction
+| CompletedAction
+| ByQueryAction;
+
+const filterAll = (): AllAction => ({ type: 'filter/ALL' });
+const filterActive = (): ActiveAction => ({ type: 'filter/ACTIVE' });
+const filterCompleted = (): CompletedAction => ({ type: 'filter/COMPLETED' });
+const filterByQuery = (query: string): ByQueryAction => ({
+  type: 'filter/QUERY',
+  payload: query,
+});
+
+export const actions = {
+  filterAll,
+  filterActive,
+  filterCompleted,
+  filterByQuery,
+};
 
-const filterReducer = () => {
-  return {
+const filterReducer = (
+  state = {
     query: '',
     status: 'all',
-  };
+  },
+  action: Action,
+) => {
+  switch (action.type) {
+    case 'filter/ALL':
```

Comment:
it would look a bit simpler with one action `filter/STATUS` that accepts payload (status) that should be set (same as QUERY action)